k9mail-library/src/test/java/com/fsck/k9/mail/filter/LineWrapOutputStreamTest.java

```diff
@@ -0,0 +1,233 @@
+            "rapid application development in many areas on most platforms.   The Python " +
+            "interpreter and the extensive standard library are freely available in source or " +
+            "binary form for all major platforms from the Python Web site, https://www.python" +
+            ".org/, and may be freely distributed. The same site also contains distributions of " +
+            "and pointers to many free third party Python modules, programs and tools, and " +
+            "additional documentation.   The Python interpreter is easily extended with new " +
+            "functions and data types implemented in C or C++ (or other languages callable from " +
+            "C). Python is also suitable as an extension language for customizable applications. " +
+            "  This tutorial introduces the reader informally to the basic concepts and features " +
+            "of the Python language and system. It helps to have a Python interpreter handy for " +
+            "hands-on experience, but all examples are self-contained, so the tutorial can be " +
+            "read off-line as well.   For a description of standard objects and modules, see The " +
+            "Python Standard Library. The Python Language Reference gives a more formal " +
+            "definition of the language. To write extensions in C or C++, read Extending and " +
+            "Embedding the Python Interpreter and Python/C API Reference Manual. There are also " +
+            "several books covering Python in depth.   This tutorial does not attempt to be " +
+            "comprehensive and cover every single feature, or even every commonly used feature. " +
+            "Instead, it introduces many of Python’s most noteworthy features, and will give you " +
+            "a good idea of the language’s flavor and style. After reading it, you will be able " +
+            "to read and write Python modules and programs, and you will be ready to learn more " +
+            "about the various Python library modules described in The Python Standard Library.  " +
+            " The Glossary is also worth going through.";
+
+    @Test
+    public void testWithForTrailingSpacePreservation() throws IOException {
+        Buffer baos = new Buffer();
+        writeToLineWrapStream(20, baos);
+        assertEquals("Mismatch with wrap length of 20", EXPECTED_WRAPPED_AT20, baos.readUtf8());
+        writeToLineWrapStream(40, baos);
+        assertEquals("Mismatch with wrap length of 40", EXPECTED_WRAPPED_AT40, baos.readUtf8());
+        writeToLineWrapStream(60, baos);
+        assertEquals("Mismatch with wrap length of 60", EXPECTED_WRAPPED_AT60, baos.readUtf8());
+        writeToLineWrapStream(80, baos);
+        assertEquals("Mismatch with wrap length of 80", EXPECTED_WRAPPED_AT80, baos.readUtf8());
+    }
+
+    private static void writeToLineWrapStream(int MAX_BYTES_PER_LINE, Buffer baos)
```

Comment:
Please put helper methods at the end of the file, so test methods come first and are grouped together.